backend/tracim_backend/lib/core/user.py

```diff
@@ -670,12 +681,66 @@ def authenticate(self, password: str, login: str, ldap_connector: "Connector" =
 
         raise AuthenticationFailed("Auth mechanism for this user is not activated")
 
+    def saml_authenticate(
+        self,
+        user: typing.Optional[User],
+        user_id: str,
+        name: str,
+        mail: str,
+    ) -> User:
+        """
+        Authenticate with ldap, return authenticated user or raise Exception
+        like WrongAuthTypeForUser, WrongLDAPCredentials, UserDoesNotExist
+        or UserAuthenticatedIsNotActive
+        :param user: user to check,, can be none if user not found, will try
+         to create new user if none but ldap auth succeed
+        :param user_id: user_id of the user
+        :param name: name extracted from saml attributes
+        :param mail: mail extracted from saml attributes
+        """
+        auth_type = AuthType.SAML
+
+        # TODO - M.L - 2023/09/06 - Change this to get saml response and isolate attributes
+
+        if not user:
+            try:
+                user = self.get_one_by_login(user_id)
+            except UserDoesNotExist:
+                _ = self.create_user(
```

Comment:
Here you need to check external_id in order to update an existing account or to create a new one. I suggest you to add a TODO so we can merge and find back where to fix #6258

Reply:
By the way, I'm afraid this wont work. The SAML IdP can't know our internal `user_id`.

Reply:
done in https://github.com/tracim/tracim/pull/6259